File: .UnusedAlternativeMethods/GifResize.py
import os
import sys
from PIL import Image
import base64
from io import BytesIO
import io

''' Required for .gif resizing in CustomPanelForSendingPictures.js '''
args = sys.argv
args = args[1:] # First element of args is the file name
if len(args) < 3:
	print('You have not passed commands! Start parameters: inputFile width height')
else:
	durs = []
	trnsprn = []
	i = 0
	img = Image.open(args[0])
	frames = []
	try:
		while (True):
			img.seek(i)
			frame = Image.new('RGBA', img.size)
			frame.paste(img, (0,0), img.convert('RGBA'))
			durs.append(img.convert('RGBA').info['duration'])
			try:
				trnsprn.append(img.info['transparency'])
			except:
				trnsprn.append(255)
			frame.thumbnail((int(args[1]),int(args[2])))
			frame.info = img.info
			frames.append(frame)
			i += 1
	except EOFError:
		pass
		buffer = BytesIO()
		frames[0].save(buffer,format="GIF", save_all = True,append_images = frames[1:] ,disposal = 2, loop = 0, transparency=trnsprn[0], optimize=False, duration=durs)
		#frames[0].save("Output.gif", save_all = True,append_images = frames[1:] ,disposal = 2, loop = 0, transparency=trnsprn[0], optimize=False, duration=durs)
		print("data:image/gif;base64," + base64.b64encode(buffer.getvalue()).decode())
exit()

'''In Discord plugin: child_process_.execSync(`python "${picturesPath}ResizeGif.py" "${_path}" ${newWidth} ${newHeight}`)'''

File: .UnusedAlternativeMethods/GifResize.js
const gifResize = require('@gumlet/gif-resize');
const fs = require("fs");
const path = require("path");

/* Required for .gif resizing in CustomPanelForSendingPictures.js */
/* Uses following modules: @gumlet, bin-wrapper, cross-spawn, end-of-stream, execa, get-stream, gifsicle, human-signals, import-lazy, is-gif, is-stream, isexe, merge-stream. mimic-fn, npm-run-path, once, onetime, path-key, pump, shebang-command, shebang-regex, signal-exit, strip-final-newline, which, wrappy */
/* For compile: pkg -t node8.17.0-win-x86,node8.17.0-macos-x64,node8.17.0-linux-x64 "GifResize.js" */

var arguments = process.argv; // Get commands line arguments
if(arguments.length < 5) { console.warn('You have not passed commands! Start parameters: inputFile width height'); return process.exit(); }
var inputFile = arguments[2];
var Width = Number(arguments[3]);
var Height = Number(arguments[4]);
var Command = arguments[5] ? arguments[5].toLowerCase() : 'log';

if(!Number.isInteger(Width) || !Number.isInteger(Height)) { console.warn(`Width and Height should be integers!`); return process.exit(); }
if(path.extname(inputFile).toLowerCase() != '.gif') { console.warn(`Input file should be gif!`); return process.exit(); }
if(!fs.existsSync(inputFile)) { console.warn(`File doesn't exist!`); return process.exit(); }

const Buff = fs.readFileSync(inputFile); // Read file
const fileOutput = 'CustomPanelForSendingPictures[!fileOutput!]'; // Output file name
gifResize({
	width: Width,
	height: Height
})(Buff)
.then(data => { // In log case received message need be read from utf-8
	if('File'.toLowerCase() == Command) { fs.writeFileSync(`${fileOutput}.gif`, data); }
	if('JSON'.toLowerCase() == Command) { fs.writeFileSync(`${fileOutput}.json`, `"data:image/gif;base64,${data.toString('base64')}"`); }
	if('Log'.toLowerCase() == Command) { console.log(`data:image/gif;base64,${data.toString('base64')}`); }
})
.catch((err) => { console.warn(err); return process.exit(); })
.finally((err) => { return process.exit(); });